src/sql/azdata.proposed.d.ts

```diff
@@ -26,4 +26,56 @@ declare module 'azdata' {
 		 */
 		export function registerConnectionEventListener(listener: connection.ConnectionEventListener): void;
 	}
+
+
+	export type SqlDbType = 'BigInt' | 'Binary' | 'Bit' | 'Char' | 'DateTime' | 'Decimal'
```

Comment:
this seems highly coupled to SQL Server.  is the expectation that only SQL Tools Service will implement this provider?

Reply:
That's a good question. a couple reasons for this current design:
- Data returned from DMP providers (MSSQL, PGSQL) currently implement this format
- Some types such as  DateTime have special handling (e.g. in Excel serializer) and so we need to preserve them
- Future providers should be able to convert this list of types into an appropriate output type. For example there's logic to correctly write these types in the SqlToolsService backend for JSON, and future providers (e.g. a yml provider) should really keep this up.

Overall, I believe we should keep this as-is for now. We can consider adding additional Javascript types (string, number, boolean) and support for those which is straightforward to update in the SqlToolsService. For now I'm going to "won't fix" this comment.